File: src/remove_bio.py
import re

file_path = r'c:\Users\Arlei\OneDrive\Desktop\novo projeto antigravity\src\App.jsx'

with open(file_path, 'r', encoding='utf-8') as f:
    text = f.read()

# 1. Remove import
text = re.sub(r"import \{ isBiometryAvailable.*?\n", "", text)

# 2. Remove states
text = re.sub(r"    // Biometria(?:.|\n)*?setShowBiometryInvite\(false\)\n", "", text) # this might be too greedy

text = re.sub(r"    // Biometria\n    const \[biometrySupported, setBiometrySupported\] = useState\(false\)\n    const \[showBiometryInvite, setShowBiometryInvite\] = useState\(false\)\n", "", text)

# 3. Remove useEffect and handlers
text = re.sub(r"    // Verificar suporte a biometria\n    useEffect\(\(\) => \{\n        const checkBiometry = async \(\) => \{.+?finally \{\n            setLoading\(false\)\n        \}\n    \}\n", "", text, flags=re.DOTALL)

# 4. Remove props from Login
text = re.sub(r"            <Login \n                biometrySupported=\{biometrySupported\} \n                onBiometricLogin=\{handleLoginWithBiometry\} \n            />", "            <Login />", text)

# 5. Remove Modal at bottom
text = re.sub(r"            \{/\* Modal de Convite para Biometria \*/\}\n            \{showBiometryInvite && \(.+?                  </div>\n                </div>\n            \)\}\n", "", text, flags=re.DOTALL)

with open(file_path, 'w', encoding='utf-8') as f:
    f.write(text)
print("Biometry removed successfully.")

File: src/remove_bio.js
const fs = require('fs');
const path = require('path');

const filePath = path.join(__dirname, 'App.jsx');
let text = fs.readFileSync(filePath, 'utf-8');

// 1. Remove import
text = text.replace(/import \{ isBiometryAvailable[\s\S]*?webauthn_helpers'\n/g, '');

// 2. Remove states
text = text.replace(/    \/\/ Biometria\n    const \[biometrySupported, setBiometrySupported\] = useState\(false\)\n    const \[showBiometryInvite, setShowBiometryInvite\] = useState\(false\)\n/g, '');

// 3. Remove useEffect and handlers
text = text.replace(/    \/\/ Verificar suporte a biometria\n    useEffect\(\(\) => \{[\s\S]+?finally \{\n            setLoading\(false\)\n        \}\n    \}\n/g, '');

// 4. Remove props from Login
text = text.replace(/            <Login \n                biometrySupported=\{biometrySupported\} \n                onBiometricLogin=\{handleLoginWithBiometry\} \n            \/>/g, '            <Login />');

// 5. Remove Modal at bottom
text = text.replace(/            \{\/\* Modal de Convite para Biometria \*\/\}\n            \{showBiometryInvite && \([\s\S]+?                  <\/div>\n                <\/div>\n            \}\)\}\n/g, '');

fs.writeFileSync(filePath, text, 'utf-8');
console.log("Biometry removed successfully via Node.");
